### ISSUE TITLE
User Joined LogiX node breaks on headless start

### ISSUE THREAD

**Issue by shadowpanther** [Frooxius/NeosPublic#1355]
When starting the world on a headless client, I get a bunch of errors. There are two kinds:
```
World Update is taking over 5 seconds to complete, breaking impulses.

   at void BaseX.UniLog.Error(string message, bool stackTrace)
   at void BaseX.UniLog.Warning(string message, bool stackTrace)
   at bool FrooxEngine.LogiX.LogixController.BeginImpulse()
   at void FrooxEngine.LogiX.Impulse.Trigger()
   at void FrooxEngine.LogiX.ProgramFlow.OnStartNode.OnStart()
   at void FrooxEngine.ComponentBase<FrooxEngine.Component>.InternalRunStartup()
   at bool FrooxEngine.UpdateManager.RunStartups()
   at bool FrooxEngine.World.RefreshStep()
   at bool FrooxEngine.World.Refresh()
   at void FrooxEngine.WorldManager.UpdateStep(double maxMilliseconds)
   at bool FrooxEngine.WorldManager.RunUpdateLoop(double maxMilliseconds)
   at void FrooxEngine.Engine.UpdateStep(double maxMilliseconds)
   at void FrooxEngine.Engine.RunUpdateLoop(double maxMilliseconds)
   at void NeosHeadless.Program.UpdateLoop()
   at void System.Threading.ThreadHelper.ThreadStart_Context(object state)
   at void System.Threading.ThreadHelper.ThreadStart()
```
Then there's the join message `User Joined Z36 - Cyber Laboratory. Username: Shadow Headless, UserID: U-Shadow-Headless, MachineID: uczxo8jttuuku__mcjreqw`. And after that:
```
World Update is taking over 5 seconds to complete, breaking impulses.

   at void BaseX.UniLog.Error(string message, bool stackTrace)
   at void BaseX.UniLog.Warning(string message, bool stackTrace)
   at bool FrooxEngine.LogiX.LogixController.BeginImpulse()
   at void FrooxEngine.LogiX.Impulse.Trigger()
   at async Task FrooxEngine.LogiX.ProgramFlow.DelayNode.DelayImpulse(float delay)
   at void System.Runtime.CompilerServices.AsyncMethodBuilderCore+MoveNextRunner.InvokeMoveNext(object stateMachine)
   at void System.Runtime.CompilerServices.AsyncMethodBuilderCore+MoveNextRunner.Run()
   at void System.Threading.Tasks.AwaitTaskContinuation.InvokeAction(object state)
   at void System.Threading.Tasks.AwaitTaskContinuation.RunCallback(ContextCallback callback, object state, ref Task currentTask)
   at void System.Threading.Tasks.SynchronizationContextAwaitTaskContinuation.Run(Task ignored, bool canInlineContinuationTask)
   at void System.Threading.Tasks.Task.FinishContinuations()
   at void System.Threading.Tasks.Task.FinishStageThree()
   at bool System.Threading.Tasks.Task<System.Threading.Tasks.VoidTaskResult>.TrySetResult(VoidTaskResult result)
   at void System.Runtime.CompilerServices.AsyncTaskMethodBuilder<System.Threading.Tasks.VoidTaskResult>.SetResult(VoidTaskResult result)
   at void System.Runtime.CompilerServices.AsyncTaskMethodBuilder<System.Threading.Tasks.VoidTaskResult>.SetResult(Task<VoidTaskResult> completedTask)
   at void System.Runtime.CompilerServices.AsyncTaskMethodBuilder.SetResult()
   at async Task FrooxEngine.Worker.DelayTimeSpan(TimeSpan timespan)
   at void System.Runtime.CompilerServices.AsyncMethodBuilderCore+MoveNextRunner.InvokeMoveNext(object stateMachine)
   at void System.Runtime.CompilerServices.AsyncMethodBuilderCore+MoveNextRunner.Run()
   at void FrooxEngine.ToWorld.OnCompleted(Action continuation)+(object u) => { }
   at int FrooxEngine.CoroutineManager.ExecuteWorldQueue(double deltaTime)
   at bool FrooxEngine.World.RefreshStep()
   at bool FrooxEngine.World.Refresh()
   at void FrooxEngine.WorldManager.UpdateStep(double maxMilliseconds)
   at bool FrooxEngine.WorldManager.RunUpdateLoop(double maxMilliseconds)
   at void FrooxEngine.Engine.UpdateStep(double maxMilliseconds)
   at void FrooxEngine.Engine.RunUpdateLoop(double maxMilliseconds)
   at void NeosHeadless.Program.UpdateLoop()
   at void System.Threading.ThreadHelper.ThreadStart_Context(object state)
   at void System.Threading.ThreadHelper.ThreadStart()
```

Looks like those are caused by User Joined node, because it breaks (becomes red) after this. I've tested with smaller script, and with the User Joined node by itself. See https://discord.com/channels/402159838827905024/501866888964341790/778403065212305408 for screenshots.
Weirdly, User Joined node actually works (generates impulses) in this state.

**Comment by Frooxius** [comment]
This doesn't look like the node itself is breaking, but rather that it's taking way too long to process whatever is it doing, which triggers LogiX's mechanism to prevent locking up for too long. Do you get a hitch when users joins? How complex is the world they're joining?

**Comment by ProbablePrime** [comment]
Pasting from Discord:
![](https://cdn.discordapp.com/attachments/501866888964341790/778411833106759740/2020-11-18_03.09.36.webp)
![](https://cdn.discordapp.com/attachments/501866888964341790/778415502863237191/2020-11-18_03.24.23.webp)

It looks like it doesn't matter.

**Comment by shadowpanther** [comment]
@Frooxius It's not when the user joins, it's only when headless opens the world (and joins) on start. When other users join, it all works, but the node still shows red.

**Comment by Frooxius** [comment]
Forgot to close this, should be fixed in 2020.11.21.1319. Please reopen if the issue persists.
